src/translib/transformer/xfmr_vlan.go

```diff
@@ -0,0 +1,167 @@
+package transformer
+
+import (
+        "errors"
+        log "github.com/golang/glog"
+        "strings"
+        "strconv"
+        "translib/ocbinds"
+        "translib/db"
+        "github.com/openconfig/ygot/ygot"        
+)
+
+func init() {
+    XlateFuncBind("DbToYang_netinst_vlans_subtree_xfmr", DbToYang_netinst_vlans_subtree_xfmr)
+}
+
+
+func getUriAttributes(inParams XfmrParams) (*ocbinds.OpenconfigNetworkInstance_NetworkInstances_NetworkInstance_Vlans, string, string, uint16, error) {
+
+    var err error
+    var vlanName string
+    pathInfo := NewPathInfo(inParams.uri)
+    niName := pathInfo.Var("name")
+    vlanIdStr := pathInfo.Var("vlan-id")
+    var vlanId uint16
+
+    targetUriPath, _ := getYangPathFromUri(pathInfo.Path)
+
+    if len(niName) == 0 {
+        return nil, "", "", 0, errors.New("Network-instance name is missing")
+    }
+
+    deviceObj := (*inParams.ygRoot).(*ocbinds.Device)
+    netInstsObj := deviceObj.NetworkInstances
+
+    if netInstsObj.NetworkInstance == nil {
+        return nil, "", "", 0, errors.New("Network-instances container missing")
+    }
+
+    netInstObj := netInstsObj.NetworkInstance[niName]
+    if netInstObj == nil {
```

Comment:
for top level GET and network instance level GET, this may come as nil, BuildEmptyTree is needed here